glib/src/variant.rs

```diff
@@ -842,13 +878,15 @@ impl fmt::Debug for Variant {
 
 impl fmt::Display for Variant {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let serialized: GString = unsafe {
-            from_glib_full(ffi::g_variant_print(
-                self.to_glib_none().0,
-                false.into_glib(),
-            ))
-        };
-        f.write_str(&serialized)
+        f.write_str(&self.print(false))
```

Comment:
> Should this be changed to print(true)?

I didn't want to change the existing behaviour just in case, but I do agree that including types would make it more consistent.

> That makes it so the string can always be parsed again.

Are there cases where the un-annotated string would actually fail to parse? My assumption was that `v.to_string().parse::<Variant>()` would still be expected to work, but it might be a lossy conversion where you end up with slightly different inferred types for things like integers (which is messy and not ideal regardless)

Reply:
> Are there cases where the un-annotated string would actually fail to parse?

It won't throw an error from `g_variant_parse` but the types will be incorrect, for example try to change the unit test to `Variant::parse(None, &a.to_string())` and it will then fail the equality check

Reply:
@bilelmoussaoui

Reply:
I also think this should be changed to `print(true)`.